### PR TITLE
Adds columns kwarg to plot methods

### LINKED ISSUE #5729: Add `column` kwarg to all timeseries source `plot` methods
In sunpy 3.1 I added the ability to plot a subset of the columns in a `GenericTimeseries`: https://docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html#sunpy.timeseries.GenericTimeSeries.plot

It would be good to roll a similar change out to all the timeseries sources.

### PR DESCRIPTION
**Pull request by Krish2208**
## PR Description
Adds column kwarg to timeseries source plot methods. 
Fixes #5729

### CODE REVIEW

File: sunpy/timeseries/sources/eve.py [hunk 1]
```diff
@@ -78,15 +80,18 @@ def plot(self, axes=None, **kwargs):
...
+            axes = self.to_dataframe()[column].plot(subplots=True, sharex=True, **kwargs)
         for i, ax in enumerate(axes):
             if predefined_axes:
-                ax.plot(self._data[self._data.columns[i]],
-                        color=colors[i % len(colors)],
-                        label=self._data.columns[i])
+                if i in column:
+                    ax.plot(self._data[self._data.columns[i]],
+                            color=colors[i % len(colors)],
+                            label=self._data.columns[i])
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py:94]: How does this work?

Comment (Krish2208) [reply sunpy/timeseries/sources/eve.py:94]: Sorry, my bad I messed up there. I am on it...

**Comment by Krish2208** [comment]
Sorry, I was unable to work due to certain circumstances.
Earlier I forget to render the names according to the column so I added that too.

File: sunpy/timeseries/sources/eve.py [hunk 2]
```diff
@@ -74,22 +76,25 @@ def plot(self, axes=None, **kwargs):
...
         predefined_axes = False
+        if column is None:
+            column = self.to_dataframe().columns
         if isinstance(axes, np.ndarray):
             predefined_axes = True
         elif axes is None:
-            axes = self.to_dataframe().plot(subplots=True, sharex=True, **kwargs)
-        for i, ax in enumerate(axes):
+            axes = self.to_dataframe()[column].plot(subplots=True, sharex=True, **kwargs)
+        for i, name in enumerate(self.to_dataframe()[column]):
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py:90]: is `name` used somewhere?

File: sunpy/timeseries/sources/eve.py [hunk 3]
```diff
@@ -74,22 +76,25 @@ def plot(self, axes=None, **kwargs):
...
         """
         self._validate_data_for_plotting()
-        names = ("Flux \n 0.1-7nm", "Flux \n 18nm", "Flux \n 26nm", "Flux \n 30nm", "Flux \n 36nm")
+        column_names = {"QD": "Flux \n 0.1-7nm", "CH_18": "Flux \n 18nm",
+                        "CH_26": "Flux \n 26nm", "CH_30": "Flux \n 30nm", "CH_36": "Flux \n 36nm"}
         colors = ('tab:blue', 'tab:orange', 'tab:green', 'tab:red', 'tab:purple',
                   'tab:brown', 'tab:pink', 'tab:gray', 'tab:olive', 'tab:cyan')
         predefined_axes = False
+        if column is None:
+            column = self.to_dataframe().columns
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py:85]: I am not sure if this is done in all the other functions but we should cast to a dataframe once.

File: sunpy/timeseries/sources/fermi_gbm.py [hunk 4]
```diff
@@ -59,14 +59,16 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
     # Class attribute used to specify the source class of the TimeSeries.
     _source = 'gbmsummary'
 
-    def plot(self, axes=None, **kwargs):
+    def plot(self, axes=None, column=None, **kwargs):
```
Comment (nabobalis) [inline sunpy/timeseries/sources/fermi_gbm.py:62]: The original issue says its `columns` and not `column`, can you check which one is corrrect?

**Comment by nabobalis** [comment]
It will be easier if you try to run the time series tests or even the sample data example (I think we plot most of the timeseries there). It should make debugging easier.

File: sunpy/timeseries/sources/fermi_gbm.py (hunk 4 above)
Comment (Krish2208) [reply sunpy/timeseries/sources/fermi_gbm.py:62]: In the title of the issue it is `column`.

File: sunpy/timeseries/sources/eve.py (hunk 3 above)
Comment (Krish2208) [reply sunpy/timeseries/sources/eve.py:85]: Sorry, I don't follow, what exactly is required?

File: sunpy/timeseries/sources/eve.py (hunk 2 above)
Comment (Krish2208) [reply sunpy/timeseries/sources/eve.py:90]: I thought of using it to set the label but I was uncertain about it.
Okk I will make the changes accordingly

**Comment by Krish2208** [comment]
I don't know what's the problem but when I run the tests it takes the installed `sunpy` package and not the local development source.

File: sunpy/timeseries/sources/fermi_gbm.py (hunk 4 above)
Comment (nabobalis) [reply sunpy/timeseries/sources/fermi_gbm.py:62]: Can you check this again, what if the title was wrong?

File: sunpy/timeseries/sources/eve.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/timeseries/sources/eve.py:85]: There are multiple calls to convert the time series to a data frame. This should only happen once.

**Comment by nabobalis** [comment]
> I don't know what's the problem but when I run the tests it takes the installed `sunpy` package and not the local development source.

Did you install the local development version?

**Comment by Krish2208** [comment]
Ohh I didn't do it earlier, now I did it, thank you.

File: sunpy/timeseries/sources/rhessi.py [hunk 5]
```diff
@@ -192,12 +196,14 @@ def peek(self, title="RHESSI Observing Summary Count Rate", **kwargs):
         ----------
         title : `str`
             The title of the plot. Defaults to "RHESSI Observing Summary Count Rate".
+        columns : list[str], optional
+            If provided, only plot the specified columns.
```
Comment (nabobalis) [inline sunpy/timeseries/sources/rhessi.py:200]: This is in the wrong order.

**Comment by nabobalis** [comment]
Do all the plots still look the same then?

I would also like some unit tests just to check the columns code is at least passed and doesn't error. They do not need to be new figure tests.

**Comment by Krish2208** [comment]
> Do all the plots still look the same then?

I didn't add columns to some of the plot methods as

- `NoRHTimeSeries` it only have one column, so I thought that `columns` kwarg won't be neccessary.
- `NOAAPredictIndicesTimeSeries` & `NOAAIndicesTimeSeries` because right now also they are plotting specific columns.

If any changes are required, I will do that.

> I would also like some unit tests just to check the columns code is at least passed and doesn't error. They do not need to be new figure tests.

Ok, I will write some unit tests.

**Comment by nabobalis** [comment]
> I didn't add columns to some of the plot methods as
> 
> * `NoRHTimeSeries` it only have one column, so I thought that `columns` kwarg won't be neccessary.
> * `NOAAPredictIndicesTimeSeries` & `NOAAIndicesTimeSeries` because right now also they are plotting specific columns.
> 

I don't see the harm in adding the keyword and just doing nothing with it.

File: changelog/6056.feature.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Adds `columns` kwarg to plot method to timeseries sources.
```
Comment (nabobalis) [inline changelog/6056.feature.rst:1]:
```suggestion
Added a ``columns`` keyword to each plot method for all `sunpy.timeseries.GenericTimeSeries` sources.
```

File: sunpy/timeseries/sources/eve.py [hunk 7]
```diff
@@ -57,14 +57,16 @@ class ESPTimeSeries(GenericTimeSeries):
...
-    def plot(self, axes=None, **kwargs):
+    def plot(self, axes=None, columns=None, **kwargs):
         """
         Plots the EVE ESP Level 1 timeseries data.
 
         Parameters
         ----------
         axes : numpy.ndarray of `matplotlib.axes.Axes`, optional
             The axes on which to plot the TimeSeries.
+        columns : list[str], optional
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py:68]: Note to self: Check how this renders on RTD.

File: sunpy/timeseries/sources/eve.py [hunk 8]
```diff
@@ -215,8 +222,8 @@ def peek(self, column=None, **kwargs):
 
         Parameters
         ----------
-        column : `str`, optional
-            The column to display. Defaults to ``None``, so it will display all.
+        columns : `str`, optional
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py:225]: So this is a string but the plot method wants a list of strings, we should unify this somehow.

File: sunpy/timeseries/sources/goes.py [hunk 9]
```diff
@@ -146,7 +146,7 @@ def observatory(self):
         return None
 
     @peek_show
-    def peek(self, title="GOES X-ray flux", **kwargs):
+    def peek(self, columns=["xrsa", "xrsb"], title="GOES X-ray flux", **kwargs):
```
Comment (nabobalis) [inline sunpy/timeseries/sources/goes.py:149]: Why could this be set to None and then have it plot the two columns by default?

File: sunpy/timeseries/sources/noaa.py [hunk 10]
```diff
@@ -67,6 +67,8 @@ def plot(self, axes=None, plot_type='sunspot SWO', **kwargs):
             The axes on which to plot the TimeSeries.
         plot_type : {``"sunspot SWO"``, ``"sunspot RI"``, ``"sunspot compare"``, ``"radio"``, ``"geo"``}, optional
             The type of plot required. Defaults to ``"sunspot SWO"``.
+        columns : *
```
Comment (nabobalis) [inline sunpy/timeseries/sources/noaa.py:70]: This shouldn't be a *

File: sunpy/timeseries/tests/test_timeseriesbase.py [hunk 11]
```diff
@@ -951,6 +951,39 @@ def test_timeseries_array():
         ts = sunpy.timeseries.TimeSeries(data, {})
         assert isinstance(ts, sunpy.timeseries.GenericTimeSeries)
 
+# =============================================================================
+# Test Source Plot columns keyword argument
+# =============================================================================
+
+
```
Comment (nabobalis) [inline sunpy/timeseries/tests/test_timeseriesbase.py:958]: Are these the only 4 sources that were missing?

**Comment by nabobalis** [comment]
Can you fix the failing tests?

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @Krish2208, sorry it took so long!

**Pull request merged**